src/argilla_server/validators/records.py

```diff
@@ -0,0 +1,83 @@
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import copy
+from abc import ABC, abstractmethod
+from typing import Dict, Union
+
+from argilla_server.models import Dataset
+from argilla_server.schemas.v1.records import RecordCreate, RecordUpdate
+
+
+class RecordValidatorBase(ABC):
```

Comment:
Why do you need `ABC`?

Reply:
This is the standard way to define [abstract classes](https://docs.python.org/3/library/abc.html) in python. Instead of defining the method + raise NotImplementedError, abstract method uses the `@abstractmethod` decorator + `pass` or `...` in the body